com.unity.render-pipelines.universal/Runtime/2D/Passes/Utility/RendererLighting.cs

```diff
@@ -586,33 +585,36 @@ private static Material CreateLightMaterial(Renderer2DData rendererData, Light2D
 
         private static Material GetLightMaterial(this Renderer2DData rendererData, Light2D light, bool isVolume)
         {
-            uint materialIndex = GetLightMaterialIndex(light, isVolume);
+            var materialIndex = GetLightMaterialIndex(light, isVolume);
 
-            if (rendererData.lightMaterials[materialIndex] == null)
-                rendererData.lightMaterials[materialIndex] = CreateLightMaterial(rendererData, light, isVolume);
+            if (!rendererData.lightMaterials.TryGetValue(materialIndex, out var material))
+            {
+                material = CreateLightMaterial(rendererData, light, isVolume);
+                rendererData.lightMaterials[materialIndex] = material;
+            }
 
             return rendererData.lightMaterials[materialIndex];
```

Comment:
We could use the output of `TryGetValue` above instead of doing a lookup again here.

Reply:
fixed